### PR TITLE
Admin: Correct logic flow for nonce and capability checks in post.php

### PR DESCRIPTION
**Pull request by liaisontw**
Description
This PR addresses a logic flaw in wp-admin/post.php during the post-quickdraft-save action. Previously, the code attempted to fetch the post object and require dashboard includes before verifying the nonce or checking user capabilities.

Changes included in this PR:

Reordered Logic: Nonce and parameter validation now occur at the beginning of the switch case, ensuring unauthorized or malformed requests are rejected before any heavy processing (like get_post()) occurs.

PHP 8.2+ Compatibility: Implemented null coalescing operators (??) when accessing $_REQUEST parameters to prevent Undefined array key notices.

Unit Tests: Added PHPUnit test cases in Tests_Admin_Post_QuickDraftSave to verify that invalid nonces or missing parameters are handled correctly without triggering PHP warnings.

Trac ticket: https://core.trac.wordpress.org/ticket/65052

Use of AI Tools
AI assistance: Yes
Tool(s): Gemini
Model(s): Gemini 3 Flash
Used for: Refining the PHPUnit test structure and drafting the technical description for the Pull Request. The final code logic, security implications, and test assertions were reviewed and verified by me to ensure compliance with WordPress Coding Standards.

This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.

### CODE REVIEW

File: src/wp-admin/post.php [hunk 1]
```diff
@@ -73,12 +73,14 @@
 	case 'post-quickdraft-save':
 		// Check nonce and capabilities.
 		$nonce     = $_REQUEST['_wpnonce'];
+		$post_id   = absint( $_REQUEST['post_ID'] ?? 0 );
```
Comment (apermo) [inline src/wp-admin/post.php]: `absint()` is likely wrong here.

Likely you want `intval()` or an `(int)` typecast.

`absint()` returns the positive value for any negative int, while there should be no negative post IDs under normal circumstances, they are not impossible, and `absint()` could be a breaking change.

File: tests/phpunit/tests/admin/wpPost.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	 */
+	public function test_post_quickdraft_save_happy_path() {
+		$post_id = self::factory()->post->create( array( 'post_status' => 'draft' ) );
+		$nonce   = wp_create_nonce( 'add-post' );
+
+		$_REQUEST['_wpnonce'] = $nonce;
+		$_REQUEST['post_ID']  = $post_id;
+
+		$nonce_req = $_REQUEST['_wpnonce'] ?? '';
+		$id_req    = absint( $_REQUEST['post_ID'] ?? 0 );
```
Comment (apermo) [inline tests/phpunit/tests/admin/wpPost.php]: same as above.

File: tests/phpunit/tests/admin/wpPost.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	 * @ticket 65052
+	 * test post quickdraft save missing nonce
+	 */
+	public function test_post_quickdraft_save_missing_nonce() {
+		$_REQUEST['action'] = 'post-quickdraft-save';
+		unset( $_REQUEST['_wpnonce'] ); // invliad nonce
+		$_REQUEST['post_ID'] = 0;
+
+		$nonce   = $_REQUEST['_wpnonce'] ?? '';
+		$post_id = absint( $_REQUEST['post_ID'] ?? 0 );
```
Comment (apermo) [inline tests/phpunit/tests/admin/wpPost.php]: see above

File: tests/phpunit/tests/admin/wpPost.php [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+	/**
+	 * @ticket 65052
+	 * test post quickdraft save invalid all
+	 */
+	public function test_post_quickdraft_save_invalid_all() {
+		$_REQUEST['_wpnonce'] = 'invalid_nonce';
+		$_REQUEST['post_ID']  = -1; // invalid ID
+
+		$nonce   = $_REQUEST['_wpnonce'] ?? '';
+		$post_id = absint( $_REQUEST['post_ID'] ?? 0 );
```
Comment (apermo) [inline tests/phpunit/tests/admin/wpPost.php]: see above

**Review by apermo — CHANGES_REQUESTED** [review]

File: src/wp-admin/post.php (hunk 1 above)
Comment (liaisontw) [reply src/wp-admin/post.php]: Thanks, @apermo. You're right—there should be no negative post IDs under normal circumstances, but they are not impossible in a database context. An (int) typecast is indeed better than absint() to prevent unintended side effects where a negative input could be transformed into a valid positive ID.

I have updated the patch to use (int) and added an explicit check for $post_id > 0 before calling get_post(). This approach is based on the interaction between the frontend UI and backend processing in the Quick Draft workflow:

Frontend Design Intent: In wp_dashboard_quick_press() (wp-admin/includes/dashboard.php), the post_ID hidden field is initialized to 0 for new drafts and populated with a positive integer once a draft is saved:

`<input type="hidden" name="action" id="quickpost-action" value="post-quickdraft-save" />
<input type="hidden" name="post_ID" value="<?php echo $post_ID; ?>" />
`
The post-quickdraft-save action is specifically designed to update an existing entity.

Explicit vs. Implicit ID Handling: As defined in get_post(), passing a falsy value (like 0) triggers a fallback that returns the current global $post object. In this dashboard context, doing so would be a logic error. By enforcing $post_id > 0, we ensure the code returns null for new or invalid requests, preventing it from inadvertently modifying the global post state.

Consistency and Database Standards: Per wp-admin/includes/schema.php, the ID is defined as bigint(20) unsigned NOT NULL auto_increment. Using (int) typecasting is consistent with established core practices for ID sanitization, and the explicit $post_id > 0 check aligns with the database's positive-only primary key structure.

Security and Reliability: By avoiding absint(), we eliminate the risk of "ID collision" where a malicious negative input might target a legitimate positive post ID.

I’ve also updated the PHPUnit tests to verify that 0 and negative values are handled safely without falling back to any existing global post state.

Thanks again for the detailed review.

File: tests/phpunit/tests/admin/wpPost.php (hunk 2 above)
Comment (liaisontw) [reply tests/phpunit/tests/admin/wpPost.php]: same as above.

File: tests/phpunit/tests/admin/wpPost.php (hunk 3 above)
Comment (liaisontw) [reply tests/phpunit/tests/admin/wpPost.php]: same as above.

File: tests/phpunit/tests/admin/wpPost.php (hunk 4 above)
Comment (liaisontw) [reply tests/phpunit/tests/admin/wpPost.php]: same as above.
